### PR TITLE
Add UUID generator incorporating dictionary entry keys

### PR DESCRIPTION
**Pull request by sldouglas-nist**

### CODE REVIEW

**Comment by ajnelson-nist** [comment]
CI should finish before this merges.

File: case_utils/inherent_uuid.py
Comment (kchason) [inline case_utils/inherent_uuid.py:100]: Why accept `*args` and `**kwargs` if they're silently ignored?

**Review by kchason — APPROVED** [review]

File: case_utils/inherent_uuid.py
Comment (ajnelson-nist) [reply case_utils/inherent_uuid.py:100]: Fair point.  These were my suggestion, so I'll field the question.

This addresses a syntax in Python's function definitions that frequently looks ambiguous to me, where a kind-of-positional spelling can be used at call time.

I've found it helpful to have at least `*args` always, as a reminder that there's a point in the argument list where the arguments are no longer positional.  If it comes at the end, great!  All arguments are positional.  `**kwargs` tends to come along as a matter of habit - if there's `*args`, I tend to want `**kwargs` as well.

I also include `*args` and `**kwargs` in class methods when I know subclasses are going to override the implementations, or use `super()`.

So, by two habits, I use `*args` and `**kwargs` in function definitions, and I suggested those to @sldouglas-nist .

**Comment by ajnelson-nist** [comment]
It looks like a part of the build system post-tests needs fixing.  FWIW, it seems out of scope of the changes in this PR, so I think it would be fine to merge this PR, but I'm also fine waiting for CI to pass with another PR merged into `develop` first.

**Pull request merged**
